#184581170 Setup Docker for Backend Application

ericndungutse opened the pull request:
#### What does this PR do?
Have docker setup for backend API

#### Description of Task to be completed?
Have the following functionality configured

- Running express server using docker compose
- Creating  and running postgres using docker compose

#### How should this be manually tested?
First make sure you have docker installed. In your terminal;

1. Set environment variables. NB: **Set DB_HOST to `db`**
2. Run `docker compose up -d db` to run postgres container
3. Run `docker compose logs` to check if database is ready for connection.
4. Run `docker compose build` to build Docker image.
5. Run `docker compose up app`
6. Run `docker compose exec -it app npx sequelize-cli db:migrate` to create tables
7. Use Postman or any other api testing app to make the following request on 127.0.0.01:3000;
   - `POST /user/create` Body request {userName, email, password}
   - `GET /user/all`

#### What are the relevant pivotal tracker/Trello stories?
Story ID: [#184581170 ](https://www.pivotaltracker.com/story/show/184581170)

wilbrord2 reviewed (approved):
good

Habinezajanvier commented on docker-compose.yml:
```diff
@@ -0,0 +1,30 @@
+    image: app_amigos_ecom
+    ports:
+      - 3000:3000
+    environment:
+      - DB_NAME=amigos_ecom
+      - DB_USERNAME=postgres
+      - DB_USER_PASSWORD=test1234
+      - DB_HOST=db
```
Can we pick those data from `.env` rather than hard-coding them in `docker-compose` file

Habinezajanvier commented on docker-compose.yml:
```diff
@@ -0,0 +1,30 @@
+    image: postgres:12
+    ports:
+      - 5432:5432
+    environment:
+      - POSTGRES_DB=amigos_ecom
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=test1234
```
Sensible data should be in `.env`

Habinezajanvier commented on docker-compose.yml:
```diff
@@ -0,0 +1,30 @@
+      - DB_HOST=db
+    depends_on:
+      - db
+
```
find a way to add volumes tag on app services so that editing in dev mode can be synced in real time with changes in docker

nshutijonathan reviewed (changes requested):
Please squash your commits into one

ericndungutse committed:
chore(docker compose): setup docker for backend

 -ensure app can run in docker containers
 -create services to run express server and postgres

[Finishes #184581170]

nshutijonathan reviewed (approved):
LGTM

Habinezajanvier commented on .env.example:
```diff
@@ -9,7 +9,11 @@ DB_USER=postgres
 COVERALLS_REPO_TOKEN=b5dAPmMgZHCARyJFm6QYaOXghAF7rTD9g
+=======
+
+>>>>>>> 99a2eca... Revert "chore(docker compose): set up docker compose file"
```
You didn't finish the rebase

ericndungutse replied:
Finished.

Habinezajanvier reviewed (approved):
🥇 LGTM
